compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/transform-fire/bailout-retry/error.todo-infer-deps-on-retry.expect.md

```diff
@@ -0,0 +1,42 @@
+
+## Input
+
+```javascript
+// @inferEffectDependencies @panicThreshold(none)
+import {useRef} from 'react';
+import {useSpecialEffect} from 'shared-runtime';
+
+/**
+ * The retry pipeline disables memoization features, which means we need to
+ * provide an alternate implementation of effect dependencies which does not
+ * rely on memoization.
+ */
+function useFoo({cond}) {
+  const ref = useRef();
+  const derived = cond ? ref.current : makeObject();
+  useSpecialEffect(() => {
```

Comment:
Ah, we currently depend on memoization? I thought we only depended on reactivity analysis

Reply:
Yeah the implementation currently depends on mutability analysis + reactive scopes. `ref.current` values are a bit of an edge case for mutable ranges, but this is really an implementation todo (specifically, the retry pipeline should not use reactive scopes for dependency analysis)

For devs, this should just an unknown compiler todo bailout with "please supply your own dependency array as the compiler could not transpile this".